### PR TITLE
[Synapse] Azure Synapse Correctness

### PR DESCRIPTION
**Pull request by niks02**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [x] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by wonner** [comment]
Hi @niks02 , it will be breaking changes, you may need to add new API version or go through breaking change review process.

**Review by kashyap-kunal — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/examples/UpdateSqlPool.json [hunk 1]
```diff
@@ -18,9 +18,7 @@
         "collation": "",
         "sourceDatabaseId": "",
         "recoverableDatabaseId": "",
-        "restorePointInTime": "1970-01-01T00:00:00.000Z",
-        "createMode": "",
-        "creationDate": "1970-01-01T00:00:00.000Z"
+        "createMode": ""
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/examples/UpdateSqlPool.json]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples/UpdateSqlPool.json [hunk 2]
```diff
@@ -18,9 +18,7 @@
         "collation": "",
         "sourceDatabaseId": "",
         "recoverableDatabaseId": "",
-        "restorePointInTime": "1970-01-01T00:00:00.000Z",
-        "createMode": "",
-        "creationDate": "1970-01-01T00:00:00.000Z"
+        "createMode": ""
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples/UpdateSqlPool.json]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples/CreateIpFirewallRule.json [hunk 3]
```diff
@@ -37,6 +37,18 @@
...
       }
     },
+    "202": {
+      "body": {
+        "id": "/subscriptions/01234567-89ab-4def-0123-456789abcdef/resourceGroups/ExampleResourceGroup/providers/Microsoft.Synapse/workspaces/ExampleWorkspace/firewallRules/ExampleIpFirewallRule",
+        "type": "Microsoft.Synapse/workspaces/firewallRules",
+        "name": "ExampleIpFirewallRule",
+        "properties": {
+          "startIpAddress": "10.0.0.0",
+          "endIpAddress": "10.0.0.254",
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples/CreateIpFirewallRule.json]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/firewallRule.json [hunk 4]
```diff
@@ -119,6 +119,12 @@
             },
             "description": ""
           },
+          "202": {
+            "schema": {
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/firewallRule.json:123]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlServer.json [hunk 5]
```diff
@@ -1487,9 +1496,6 @@
     },
     "ServerVulnerabilityAssessmentProperties": {
       "description": "Properties of a server Vulnerability Assessment.",
-      "required": [
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlServer.json:1490]: Yes required, made changes.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/sqlServer.json [hunk 6]
```diff
@@ -1487,14 +1488,15 @@
     },
     "ServerVulnerabilityAssessmentProperties": {
       "description": "Properties of a server Vulnerability Assessment.",
-      "required": [
-        "storageContainerPath"
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/sqlServer.json:1491]: Yes required, made changes.

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json [hunk 7]
```diff
@@ -1208,7 +1216,8 @@
         },
         "managedResourceGroupName": {
           "description": "Workspace managed resource group. The resource group name uniquely identifies the resource group within the user subscriptionId. The resource group name must be no longer than 90 characters long, and must be alphanumeric characters (Char.IsLetterOrDigit()) and '-', '_', '(', ')' and'.'. Note that the name cannot end with '.'",
-          "type": "string"
+          "type": "string",
+          "default": ""
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json:1220]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlPool.json [hunk 8]
```diff
@@ -5361,30 +5368,36 @@
...
-          }
+          },
+          "default": "GRS"
         },
         "sourceDatabaseDeletionDate": {
           "type": "string",
           "description": "Specifies the time that the sql pool was deleted",
-          "format": "date-time"
+          "format": "date-time",
+          "readOnly": true
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlPool.json:5400]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/sqlPool.json [hunk 9]
```diff
@@ -5342,12 +5346,14 @@
...
-          "description": "Resource status"
+          "description": "Resource status",
+          "readOnly": true
         },
         "restorePointInTime": {
           "type": "string",
           "description": "Snapshot time to restore",
-          "format": "date-time"
+          "format": "date-time",
+          "readOnly": true
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/sqlPool.json:5356]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlPool.json [hunk 10]
```diff
@@ -5342,12 +5347,14 @@
...
-          "description": "Resource status"
+          "description": "Resource status",
+          "readOnly": true
         },
         "restorePointInTime": {
           "type": "string",
           "description": "Snapshot time to restore",
-          "format": "date-time"
+          "format": "date-time",
+          "readOnly": true
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlPool.json:5357]: Done

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlServer.json [hunk 11]
```diff
@@ -1237,7 +1237,8 @@
         "storageAccountSubscriptionId": {
           "format": "uuid",
           "description": "Specifies the blob storage subscription Id.",
-          "type": "string"
+          "type": "string",
+          "default": ""
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlServer.json:1241]: Yes Kunal, the only options available are to mark the property as read-only and provide a default value unless it's an exemption.

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/bigDataPool.json [hunk 12]
```diff
@@ -604,6 +615,10 @@
         "filename": {
           "type": "string",
           "description": "The filename of the library requirements file."
+        },
+        "configurationType": {
+          "type": "string",
```
[the remark this answers is not in the document]
Comment (niks02) [reply specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/bigDataPool.json:620]: I cross-checked with the code it was correct in 2021-06-01-preview version of bigdatapool, so I corrected in 2021-06-01-stable version as per the code.

**Review by kashyap-kunal — APPROVED** [review]

**Review by ruowan — APPROVED** [review]

**Pull request merged**